consensus/types/src/data_column_sidecar.rs

```diff
@@ -23,13 +25,47 @@ pub type ColumnIndex = u64;
 pub type Cell<E> = FixedVector<u8, <E as EthSpec>::BytesPerCell>;
 pub type DataColumn<E> = VariableList<Cell<E>, <E as EthSpec>::MaxBlobCommitmentsPerBlock>;
 
-/// Container of the data that identifies an individual data column.
-#[derive(
-    Serialize, Deserialize, Encode, Decode, TreeHash, Copy, Clone, Debug, PartialEq, Eq, Hash,
-)]
-pub struct DataColumnIdentifier {
+/// Container of data columns of a block root.
+#[derive(Serialize, Deserialize, Encode, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct DataColumnsByRootIdentifier {
     pub block_root: Hash256,
-    pub index: ColumnIndex,
+    pub indices: RuntimeVariableList<ColumnIndex>,
```

Comment:
nit: we don't really use the field name for encoding but we try to match the field names in the spec. I'll update this to `columns`